code/modules/admin/verbs/maprotation.dm

```diff
@@ -68,28 +64,56 @@
 			return
 
 		qdel(M)
+		var/config_file = null
+		var/json_value = list()
+		var/config = tgui_alert(usr,"Would you like to upload an additional config for this map?", "Map Config", list("Yes", "No"))
+		if(config == "Yes")
+			config_file = input("Pick file:", "Config JSON File") as null|file
+			if(isnull(config_file))
+				return
+			if(copytext("[config_file]", -5) != ".json")
+				to_chat(src, span_warning("Filename must end in '.json': [config_file]"))
+				return
+			if(!fcopy(config_file, "_maps/custom/[config_file]"))
+				return
```

Comment:
fcopy could fuck up here in case there's already some custom config_file matching this name already on the server, and BYOND could fail to overwrite it. Use `rustg_file_write()` instead.

Reply:
> fcopy could fuck up here in case there's already some custom config_file matching this name already on the server, and BYOND could fail to overwrite it. Use `rustg_file_write()` instead.

I tested this method and it seemed quite strange. So I just provided a check for the existence of the file.

Reply:
yeah i was just concerned about writing untracked files into a repository. might be benign at absolute best but hygiene is good, and writing to data could be considered some type of "expected pattern"